### PR TITLE
Add a test for checking SHAs.

### PR DESCRIPTION
**Pull request by dlorenc**
Minor refactor of notify.go to share logic for downloading/parsing the releases.json.

### CODE REVIEW

File: pkg/minikube/notify/notify.go [hunk 1]
```diff
@@ -99,16 +100,23 @@ func getJson(url string, target *releases) error {
 }
 
 func getLatestVersionFromURL(url string) (semver.Version, error) {
```
Comment (r2d4) [inline pkg/minikube/notify/notify.go]: nit: I noticed that this method will prompt you to download v0.9.0 even if you are running v0.10.0 if we haven't "released" v0.10.0 in our releases.json.  Should we do a greater than check in here?

File: pkg/minikube/notify/notify_test.go [hunk 2]
```diff
@@ -146,7 +146,7 @@ func TestMaybePrintUpdateText(t *testing.T) {
 	// test that no update text is printed if the latest version is lower/equal to the current version
 	latestVersionFromURL := "0.0.0-dev"
 	handler := &URLHandlerCorrect{
-		releases: []release{{Name: version.VersionPrefix + latestVersionFromURL}},
+		releases: []Release{{Name: version.VersionPrefix + latestVersionFromURL}},
 	}
 	server := httptest.NewServer(handler)
```
Comment (r2d4) [inline pkg/minikube/notify/notify_test.go]: Not a change in the PR but we should fix: I think you need to call close on httptest server

https://golang.org/src/net/http/httptest/server.go?s=2049:2093#L70

File: pkg/minikube/notify/notify.go [hunk 3]
```diff
@@ -39,11 +39,11 @@ const updateLinkPrefix = "https://github.com/kubernetes/minikube/releases/tag/v"
 var (
 	timeLayout                = time.RFC1123
 	lastUpdateCheckFilePath   = constants.MakeMiniPath("last_update_check")
-	githubMinikubeReleasesURL = "https://storage.googleapis.com/minikube/releases.json"
+	GithubMinikubeReleasesURL = "https://storage.googleapis.com/minikube/releases.json"
```
Comment (r2d4) [inline pkg/minikube/notify/notify.go]: should GithubMinikubeReleasesURL be a const? or in constants.go?

File: pkg/minikube/notify/notify.go (hunk 1 above)
Comment (dlorenc) [inline pkg/minikube/notify/notify.go]: Isn't this handled on line 62?

File: pkg/minikube/notify/notify.go (hunk 3 above)
Comment (dlorenc) [inline pkg/minikube/notify/notify.go]: Done.

File: pkg/minikube/notify/notify_test.go (hunk 2 above)
Comment (dlorenc) [inline pkg/minikube/notify/notify_test.go]: Done.

File: pkg/minikube/notify/notify.go (hunk 1 above)
Comment (r2d4) [inline pkg/minikube/notify/notify.go]: Yeah i see that now.  Weird, I could have sworn that I encountered this a few days ago on my personal workstation

**Pull request merged**